cli/cmd/addon_publish.go

```diff
@@ -0,0 +1,249 @@
+	AddonDeps  []string `json:"addon_deps"`
+}
+
+func runAddonPublish(ctx context.Context, cmd *cobra.Command, args []string) error {
+	tc := auth.NewTokenClient()
+	token, err := tc.GetToken()
+	if err != nil {
+		return fmt.Errorf("failed to get auth token: %w", err)
+	}
+
+	manifestPath, version := args[0], args[1]
+	manifestDir := filepath.Dir(manifestPath)
+
+	manifest, err := readManifestJSON(manifestPath)
+	if err != nil {
+		return fmt.Errorf("failed to read manifest.json: %w", err)
+	}
+
+	if manifest.Type != "addon" {
+		return errors.New("manifest.json is not of addon type")
+	}
+
+	zipPath := filepath.Join(manifestDir, manifest.PathToZip)
+	if _, err := os.Stat(zipPath); err != nil {
+		return fmt.Errorf("could not read file %s: %w", zipPath, err)
+	}
+
+	name := fmt.Sprintf("%s/%s@%s", manifest.TeamName, manifest.AddonName, version)
+	fmt.Printf("Publishing addon %s to CloudQuery Hub...\n", name)
+
+	c, err := cloudquery_api.NewClientWithResponses(getEnvOrDefault(envAPIURL, defaultAPIURL),
+		cloudquery_api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+			return nil
+		}))
+	if err != nil {
+		return fmt.Errorf("failed to create hub client: %w", err)
+	}
+
+	// create new draft version
+	if err := createNewAddonDraftVersion(ctx, c, manifest, version, manifestDir, zipPath); err != nil {
+		return fmt.Errorf("failed to create new draft version: %w", err)
+	}
+
+	// upload package
+	fmt.Println("Uploading addon...")
+	if err := uploadAddon(ctx, c, manifest, version, zipPath); err != nil {
+		return fmt.Errorf("failed to upload addon: %w", err)
+	}
+
+	// optional: mark addon as draft=false
+	finalize, err := cmd.Flags().GetBool("finalize")
+	if err != nil {
+		return err
+	}
+
+	if finalize {
+		fmt.Println("Finalizing addon version...")
+		draft := false
+		resp, err := c.UpdateAddonVersionWithResponse(ctx, manifest.TeamName, manifest.AddonName, version, cloudquery_api.UpdateAddonVersionJSONRequestBody{
+			Draft: &draft,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to finalize addon version: %w", err)
+		}
+		if resp.HTTPResponse.StatusCode > 299 {
+			return errorFromHTTPResponse(resp.HTTPResponse, resp)
+		}
+		fmt.Println("Success!")
+		fmt.Printf("%s/%s@%s is now available on the CloudQuery Hub.\n", manifest.TeamName, manifest.AddonName, version)
+		return nil
+	}
+
+	fmt.Println("Success!")
+	fmt.Println("\nNote: this addon version is marked as draft=true. You can preview and finalize it on the CloudQuery Hub, or run `cloudquery addon publish` with the --finalize flag.")
+
+	return nil
+}
+
+func createNewAddonDraftVersion(ctx context.Context, c *cloudquery_api.ClientWithResponses, manifest ManifestJSONV1, version, manifestDir, zipPath string) error {
+	if manifest.PluginDeps == nil {
+		manifest.PluginDeps = []string{}
+	}
+	if manifest.AddonDeps == nil {
+		manifest.AddonDeps = []string{}
```

Comment:
otherwise we get:
> Error: failed to create new draft version: 400: request body has an error: doesn't match schema: Error at "/addon_deps": Value is not nullable